### PR TITLE
Add multimodule documentation

### PR DESCRIPTION
**Pull request by matir91**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,36 +8,43 @@ It provides an awesome way of displaying the paged entity list and reflecting th
...
+## Factory constructors
+There's one static factory object class for each each dependency.
+- [FountainCoroutines](FountainCoroutines.md): Used to get a [`Listing`] from a Retrofit service which uses a Coroutine adapter.
+- [FountainRetrofit](FountainRetrifit.md): Used to get a [`Listing`] from a Retrofit service without using a special adapter.
+- [FountainRx](FountainRxJava2.md): Used to get a [`Listing`] from a Retrofit service which uses a RxJava2 adapter.
 
-```kotlin
-Fountain.createNetworkListing(networkDataSourceAdapter)
-```
+Each static factory has two constructors, one for each Fountain mode.
```
Comment (FerrariBruno) [inline README.md]: `...one for each of Fountain's modes`

File: README.md [hunk 2]
```diff
@@ -8,36 +8,43 @@ It provides an awesome way of displaying the paged entity list and reflecting th
...
+
+## **Network Support** 
 
-Additionally, there are some optional parameters that you can provide to the [`Listing`] creator. These parameters are specified in the [`Fountain`] section.
+It provides a [`Listing`] structure based on a common Retrofit service implementation. Note that the entities aren't saved anywhere.
+
+It can be obtained invoking `createNetworkListing` from the static factory class.
+
+It requires only one argument, a [`NetworkDataSourceAdapter`], which provides all operations that the library
```
Comment (FerrariBruno) [inline README.md]: `Only one argument is required, ...`

File: README.md [hunk 3]
```diff
@@ -8,36 +8,43 @@ It provides an awesome way of displaying the paged entity list and reflecting th
...
+## **Network Support** 
 
-Additionally, there are some optional parameters that you can provide to the [`Listing`] creator. These parameters are specified in the [`Fountain`] section.
+It provides a [`Listing`] structure based on a common Retrofit service implementation. Note that the entities aren't saved anywhere.
+
+It can be obtained invoking `createNetworkListing` from the static factory class.
+
+It requires only one argument, a [`NetworkDataSourceAdapter`], which provides all operations that the library 
+will use to handle the paging.
+The [`NetworkDataSourceAdapter`] contains two main functions: a method to check if a page can be fetched and a property to fetch it.
```
Comment (FerrariBruno) [inline README.md]: `has` instead of `contains`?

File: README.md [hunk 4]
```diff
@@ -50,15 +57,14 @@ As in the previous way, there are some optional parameters that you can specify
...
 
-The paging strategy starts with an initial service data request.
+It starts with an initial service data request.
 By default the initial data requested is three pages, but this value can be changed, in the [`PagedList.Config`](https://developer.android.com/reference/android/arch/paging/PagedList.Config.html), using the [`setInitialLoadSizeHint`](https://developer.android.com/reference/android/arch/paging/PagedList.Config.html#initialLoadSizeHint) method.
-This parameter can be set in the [`Fountain`](Listing.md) factory method. 
-When the service data comes, all data is refreshed in the [`DataSource`] using the [`CachedDataSourceAdapter`].
-Note that the [`Listing`](Listing.md) component will notify that the data changed.
+This parameter can be set in the factory constructor method. 
+When the service data comes from the service, all data is refreshed in the `DataSource` using the `CachedDataSourceAdapter`.
+Note that the `Listing` component will notify that data changed.
```
Comment (FerrariBruno) [inline README.md]: maybe `will notify that the data has changed`?

File: RetrofitNetworkDataSourceAdapter.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+```kotlin
+interface RetrofitPageFetcher<T : ListResponse<*>> {
+  fun fetchPage(page: Int, pageSize: Int): Call<T>
+}
+```
+
+This interface has only one method which is used to fetch every page.
+The library will invoke it with two parameters: 
+- `page`: The page number to be requested.
```
Comment (FerrariBruno) [inline RetrofitNetworkDataSourceAdapter.md]: maybe `The number of the page to be requested`?

File: RetrofitNetworkDataSourceAdapter.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+- `pageSize`: The page size.
+This parameter must be respected, usually the initial load page size has a different `pageSize` than the other requests.
```
Comment (FerrariBruno) [inline RetrofitNetworkDataSourceAdapter.md]: I didn't understand this sentence, maybe you can rephrase it?

File: RetrofitNetworkDataSourceAdapter.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+*If the server doesn't support a custom `pageSize`, you have to setup the [`PagedList.Config`](https://developer.android.com/reference/android/arch/paging/PagedList.Config.html) to use the same page size for all requests.
+You can do it by setting `pageSize` in the [`setInitialLoadSizeHint()`](https://developer.android.com/reference/android/arch/paging/PagedList.Config.html#initialLoadSizeHint) method.
+The `InitialLoadSizeHint` must to be a multiple of the page size.
```
Comment (FerrariBruno) [inline RetrofitNetworkDataSourceAdapter.md]: `...must be a multiple of...`

File: RetrofitNetworkDataSourceAdapter.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+This configuration can be set in the [Fountain factory](FountainRetrofit.md).*
+
+This method returns a `Call<out ListResponse<*>`, that the library will use to fetch an specific page.
```
Comment (FerrariBruno) [inline RetrofitNetworkDataSourceAdapter.md]: `...to fetch a specific page`

File: RxNetworkDataSourceAdapter.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+This interface has only one method which is used to fetch every page.
+The library will invoke it with two parameters: 
+- `page`: The page number to be requested.
+- `pageSize`: The page size.
+This parameter must be respected, usually the initial load page size has a different `pageSize` than the other requests.
+*If the server doesn't support a custom `pageSize`, you have to setup the [`PagedList.Config`](https://developer.android.com/reference/android/arch/paging/PagedList.Config.html) to use the same page size for all requests.
+You can do it by setting `pageSize` in the [`setInitialLoadSizeHint()`](https://developer.android.com/reference/android/arch/paging/PagedList.Config.html#initialLoadSizeHint) method.
+The `InitialLoadSizeHint` must to be a multiple of the page size.
```
Comment (FerrariBruno) [inline RxNetworkDataSourceAdapter.md]: `must be a multiple...`

File: RxNetworkDataSourceAdapter.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+```kotlin
+interface RxPageFetcher<T : ListResponse<*>> {
+  fun fetchPage(page: Int, pageSize: Int): Single<T>
+}
+```
+
+This interface has only one method which is used to fetch every page.
+The library will invoke it with two parameters: 
+- `page`: The page number to be requested.
```
Comment (FerrariBruno) [inline RxNetworkDataSourceAdapter.md]: maybe `The number of the page to be requested`?

File: RxNetworkDataSourceAdapter.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+- `pageSize`: The page size.
+This parameter must be respected, usually the initial load page size has a different `pageSize` than the other requests.
```
Comment (FerrariBruno) [inline RxNetworkDataSourceAdapter.md]: I didn't understand this sentence, maybe you can rephrase it?

File: RxNetworkDataSourceAdapter.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+*If the server doesn't support a custom `pageSize`, you have to setup the [`PagedList.Config`](https://developer.android.com/reference/android/arch/paging/PagedList.Config.html) to use the same page size for all requests.
+You can do it by setting `pageSize` in the [`setInitialLoadSizeHint()`](https://developer.android.com/reference/android/arch/paging/PagedList.Config.html#initialLoadSizeHint) method.
+The `InitialLoadSizeHint` must to be a multiple of the page size.
+This configuration can be set in the [Fountain factory](FountainRxs.md).*
+
+This method returns a `Single<out ListResponse<*>`, that the library will use to fetch an specific page.
```
Comment (FerrariBruno) [inline RxNetworkDataSourceAdapter.md]: `...to fetch a specific page`

File: NetworkDataSourceAdapter.md [hunk 13]
```diff
@@ -1,66 +1,45 @@
...
-  fun canFetch(page: Int, pageSize: Int): Boolean
-}
-```
 
-`NetworkDataSourceAdapter` is a `PageFetcher` with an additional method, `canFetch`, used to stop requesting pages.
-For example, if you know that the endpoint returns only 3 pages of 10 items each, and the library invokes `canFetch(page = 5, pageSize = 10)` then you should return `false`.
-You have to implement this function using the service specification.
-Sometimes the service returns the page or entity amount in the response headers or in the response body, so you have to use that information to implement this function.
+# Page Fetcher
+Page fetcher is an structure with enables the library to fetch an specific page from a service call.
```
[the remark this answers is not in the document]
Comment (matir91) [reply NetworkDataSourceAdapter.md]: It should be `... is **a** structure...`, right?

File: README.md (hunk 4 above)
Comment (matir91) [reply README.md]: shouldn't be `will notify that data has changed`?

**Pull request merged**